control-plane/csi-driver/node/src/node.rs

```diff
@@ -311,30 +314,46 @@ impl node_server::Node for Node {
         Ok(Response::new(NodeUnpublishVolumeResponse {}))
     }
 
-    /// Get volume stats method is currently not implemented,
-    /// although it's simple to do.
-    ///
-    /// TODO: Just read the data about capacity/used space
-    /// inodes/bytes from the system using the mountpoint.
+    /// Get volume stats method evaluates and returns capacity metrics
     async fn node_get_volume_stats(
         &self,
         request: Request<NodeGetVolumeStatsRequest>,
     ) -> Result<Response<NodeGetVolumeStatsResponse>, Status> {
         let msg = request.into_inner();
         trace!("node_get_volume_stats {:?}", msg);
-
-        /*
-        Ok(Response::new(NodeGetVolumeStatsResponse {
-            usage: vec![VolumeUsage {
-                total: 0 as i64,
-                unit: volume_usage::Unit::Bytes as i32,
-                available: 0,
-                used: 0,
-            }],
-        }))
-        */
-        error!("Unimplemented {:?}", msg);
-        Err(Status::new(Code::Unimplemented, "Method not implemented"))
+        if msg.volume_id.is_empty() {
+            return Err(failure!(
+                Code::InvalidArgument,
+                "Failed to stage volume: missing volume id"
+            ));
+        }
+        if msg.volume_path.is_empty() {
+            return Err(failure!(
+                Code::InvalidArgument,
+                "Failed to stage volume: missing volume path"
+            ));
+        }
+        debug!("Volume {} on {}", &msg.volume_id, &msg.volume_path);
+        let f = sys::statfs::statfs(&*msg.volume_path);
+        match f {
```

Comment:
nit: either rename `f` into something more descriptive or simply use the call directly:
```rust
match sys::statfs::statfs(&*msg.volume_path)
```